dfm/train.py

```diff
@@ -0,0 +1,182 @@
+"""Training script for training masked language models with sweeps
+
+Authors:
+    Lasse Hansen
+    Malte Højmark-Bertelsen
+
+```bash 
+PYTHONPATH="." python dfm/train.py --path_to_config_file
+```
+"""
+
+from dataclasses import dataclass
+from typing import List, Optional
+
+from transformers import (
+    AutoConfig,
+    AutoModelForPreTraining,
+    Trainer,
+    TrainingArguments,
+    AutoTokenizer,
+    DataCollatorForLanguageModeling,
+)
+from datasets import interleave_datasets
+
+from dfm.data.load import dfm_load_dataset
+from dfm.modelling.preprocess import preprocess_dataset
+from dfm.modelling.utils import read_yaml
+from dfm.modelling.model_types import MODEL_TYPES
+
+
+def main():
+    trainer = DFMTrainer(
+        pretraining_config_path="tests/test_configs/pretrain_config.yaml",
+    )
+    trainer.train()
+
+
+# TODO:
+# create configs for models (training + model)
+# create collator for T5
+@dataclass
+class DataArguments(object):
+    dataset_names: List[str]
+    interleave: bool
+    interleave_probabilities: List[float]
+    num_proc: int
+
+    @classmethod
+    def from_yaml(cls, config_path: str):
+        file = read_yaml(config_path)
+        keys = cls.__dataclass_fields__.keys()
+        yaml_data = file["data_arguments"]
+        normal_yaml_data = {key: yaml_data[key] for key in yaml_data if key in keys}
+        anormal_yaml_data = {
+            key: yaml_data[key] for key in yaml_data if key not in keys
+        }
+        tmp = cls(**normal_yaml_data)
+        for anormal_key in anormal_yaml_data:
+            setattr(tmp, anormal_key, anormal_yaml_data[anormal_key])
+        return tmp
+
+
+@dataclass
+class ModelArguments(object):
+    model_type: str
+    model_name: str
+
+    @classmethod
+    def from_yaml(cls, config_path: str):
+        file = read_yaml(config_path)
+        keys = cls.__dataclass_fields__.keys()
+        yaml_data = file["model_arguments"]
+        normal_yaml_data = {key: yaml_data[key] for key in yaml_data if key in keys}
+        anormal_yaml_data = {
+            key: yaml_data[key] for key in yaml_data if key not in keys
+        }
+        tmp = cls(**normal_yaml_data)
+        for anormal_key in anormal_yaml_data:
+            setattr(tmp, anormal_key, anormal_yaml_data[anormal_key])
+        return tmp
+
+
+class DFMTrainer:
+    """class for training a Danish Foundation Model."""
+
+    def __init__(
+        self,
+        pretraining_config_path: str,
+    ) -> None:
```

Comment:
No need to add a `None` return type

Reply:
It does make it explicit that the function does not return anything, instead of returning something, but without having a type hint.

Reply:
@KennethEnevoldsen Hm good point!